app/assets/javascripts/discourse/app/lib/sidebar/user/categories-section/category-section-link.js

```diff
@@ -127,7 +146,15 @@ export default class CategorySectionLink {
   }
 
   get currentWhen() {
-    return "discovery.unreadCategory discovery.topCategory discovery.newCategory discovery.latestCategory discovery.category discovery.categoryNone discovery.categoryAll";
+    if (this.#linkToNew && this.activeCountable) {
+      if (get(this, this.activeCountable.propertyName) > 0) {
+        return "discovery.newCategory";
+      } else {
+        return "discovery.category";
+      }
```

Comment:
I don't think e should be limiting the `currentWhen` property when there is a count. The intended behaviour here is to treat the route as active as long as one of the following category route is active regardless of the filter.

Reply:
Ok, I've changed it back.